qldpc/objects.py

```diff
@@ -17,12 +17,16 @@
 import dataclasses
 import enum
 import itertools
-from typing import Collection, Literal, Optional
+from typing import TYPE_CHECKING
 
 import networkx as nx
 
 from qldpc import abstract
 
+if TYPE_CHECKING:
+    from collections.abc import Collection
+    from typing import Literal
```

Comment:
assuming the pyproject.toml change is reverted these shouldn't need to be in a type-checking block, and if then the quotes wouldn't be necessary around the annotations below